### ISSUE TITLE
error: unknown option `name' with Git 2.5.4

### ISSUE THREAD

**Issue by ilovezfs** [Originate/git-town#902]
git-town 4.0.0 seems to be incompatible with Git 2.5.4, which is the version on OS X 10.10 Yosemite. The error during the Homebrew test block is

```
failed: brew test git-town --verbose

Stacktrace

        Testing git-town
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20170513-53236-17hwlzp.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/git-town.rb --verbose
==> git init
Initialized empty Git repository in /private/tmp/git-town-test-20170513-53237-d6mytk/.git/
==> git add testing.txt
==> git commit -m Testing!
[master (root-commit) cd43c31] Testing!
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 testing.txt
==> /usr/local/Cellar/git-town/4.0.0/bin/git-town config

2017/05/13 20:16:06 Command: git config -l --local --name
Output: error: unknown option `name'
usage: git config [<options>]
```

Should we drop support in Homebrew for <= OS X 10.10?

See https://github.com/Homebrew/homebrew-core/pull/13537.

**Comment by charlierudolph** [comment]
@kevgo thoughts? This was a change in code in `4.0.0`. The necessary option (which is actually `--name-only` was added in Git `2.6.0`.

**Comment by kevgo** [comment]
Thanks for bringing this up, @ilovezfs! 

I think the issues is in [config.go](https://github.com/Originate/git-town/blob/master/src/git/config.go#L273). @charlierudolph, let's discuss the different options and make a decision. I have ordered them in my preference:

1. We restore the older behavior back. @charlierudolph how hard is it to do that? 

2. We make Git a brew dependency for Yosemite or older. I agree with @ilovezfs that this is kind of intrusive. Probably okay since it should be also pretty much side effect free.

3. We drop support for Yosemite in Git Town 4.0. On one hand this will probably not affect too many people given that Git Town is a developer-facing tool, and developers are usually good at installing updates. On the other hand, Yosemite is only 2 years old, so it feels a bit brutal to completely drop support for it so early.

**Comment by ilovezfs** [comment]
Note that Apple has not dropped support for Yosemite yet, as https://support.apple.com/en-us/HT207797, released today, includes updates for Yosemite.

**Comment by ilovezfs** [comment]
I've dropped Yosemite support in the formula and shipped 4.0.0, but I'm happy to add back support if compatibility with < git 2.6.0 is restored at some point.

**Comment by charlierudolph** [comment]
closing in favor of #914
